src/main/java/at/hannibal2/skyhanni/features/misc/chocolatefactory/HoppityEggsLocations.kt

```diff
@@ -0,0 +1,221 @@
+
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.ReceiveParticleEvent
+import at.hannibal2.skyhanni.test.GriffinUtils.drawWaypointFilled
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.LocationUtils.clampTo
+import at.hannibal2.skyhanni.utils.LocationUtils.isInside
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzVec
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.RenderUtils.draw3DLine
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import at.hannibal2.skyhanni.utils.expand
+import net.minecraft.item.ItemStack
+import net.minecraft.util.AxisAlignedBB
+import net.minecraft.util.EnumParticleTypes
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object HoppityEggsLocations {
+
+    private val config get() = ChocolateFactoryApi.config.hoppityEggs
+
+    private val locatorItem = "EGGLOCATOR".asInternalName()
+
+    private var lastParticlePosition: LorenzVec? = null
+    private val validParticleLocations = mutableListOf<LorenzVec>()
+
+    private var drawLocations = false
+    private var firstPos = LorenzVec()
+    private var secondPos = LorenzVec()
+    private val possibleEggLocations = mutableListOf<LorenzVec>()
+
+    private var ticksSinceLastParticleFound = -1
+
+    var sharedEggLocation: LorenzVec? = null
+    var currentEggType: HoppityEggType? = null
+
+    var eggLocations: Map<IslandType, List<LorenzVec>> = mapOf()
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        resetData()
+    }
+
+    private fun resetData() {
+        validParticleLocations.clear()
+        ticksSinceLastParticleFound = -1
+        possibleEggLocations.clear()
+        firstPos = LorenzVec()
+        secondPos = LorenzVec()
+        drawLocations = false
+        sharedEggLocation = null
+        currentEggType = null
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (!isEnabled()) return
+
+        event.draw3DLine(firstPos, secondPos, LorenzColor.RED.toColor(), 2, false)
+
+        if (drawLocations) {
+            for ((index, eggLocation) in possibleEggLocations.withIndex()) {
+                val eggLabel = "§aGuess #${index + 1}"
+                event.drawWaypointFilled(
+                    eggLocation,
+                    LorenzColor.GREEN.toColor(),
+                    seeThroughBlocks = true,
+                )
+                event.drawDynamicText(eggLocation.add(y = 1), eggLabel, 1.5)
+            }
+            return
```

Comment:
this return here correct? Since the shared egg location won't work then.

Reply:
yep, if its found with solver it should not render the shared one